eslint.config.mjs

```diff
@@ -60,30 +27,108 @@ export default [
+      'no-return-assign': 'error',
+      'no-script-url': 'error',
+      'no-self-compare': 'error',
+      'no-sequences': 'error',
+      'no-shadow': 'error',
+      'no-undef-init': 'error',
+      'no-unneeded-ternary': 'error',
       'no-unused-vars': ['error', { args: 'none' }],
+      'no-useless-concat': 'error',
       'no-var': 'error',
+      'one-var': ['error', 'never'],
       'prefer-const': 'error',
       'prefer-template': 'error',
+      'radix': ['error', 'always'],
     },
   },
-
-  /* ─────────────────── Test-only overrides ─────────────────── */
+  stylistic.configs.recommended,
   {
-    name: 'Tests: presets',
-    files: ['test/**', '**/*.cy.js'],
-    ...testPreset,
+    name: 'Stylistic overrides',
+    rules: {
+      '@stylistic/array-bracket-spacing': ['error', 'never'],
+      '@stylistic/arrow-parens': ['error', 'as-needed'],
+      '@stylistic/block-spacing': ['error', 'always'],
+      '@stylistic/brace-style': ['error', '1tbs'],
+      '@stylistic/comma-dangle': ['error', 'always-multiline'],
+      '@stylistic/computed-property-spacing': ['error', 'never'],
+      '@stylistic/dot-location': ['error', 'property'],
+      '@stylistic/indent': ['error', 2, { SwitchCase: 1 }],
+      '@stylistic/object-curly-spacing': ['error', 'always'],
+      '@stylistic/operator-linebreak': [
+        'error',
+        'before',
+        {
+          overrides: {
+            '?': 'after',
+            ':': 'after',
+          },
+        },
+      ],
+      '@stylistic/padded-blocks': ['error', 'never'],
+      '@stylistic/quotes': ['error', 'single'],
+      '@stylistic/quote-props': ['error', 'consistent'],
+      '@stylistic/semi': ['error', 'always'],
+      '@stylistic/semi-spacing': ['error', { before: false, after: true }],
+      '@stylistic/space-before-function-paren': ['error', 'never'],
+      '@stylistic/space-in-parens': ['error', 'never'],
+      '@stylistic/spaced-comment': ['error', 'always'],
+      '@stylistic/template-curly-spacing': ['error', 'always'],
+      '@stylistic/wrap-iife': ['error', 'inside'],
+    },
   },
   {
-    name: 'Tests: local tweaks',
-    files: ['test/**', '**/*.cy.js'],
+    files: TEST_FILES,
+    ...mocha.configs.recommended,
+  },
+  {
+    name: 'Mocha overrides',
+    files: TEST_FILES,
+    rules: {
+      'mocha/no-mocha-arrows': 'off',
+      'mocha/no-setup-in-describe': 'off',
+      'mocha/no-pending-tests': 'error',
+    },
+  },
+  {
+    files: TEST_FILES,
+    ...chaiFriendly.configs.recommendedFlat,
+  },
+  {
+    files: TEST_FILES,
+    ...cypress.configs.recommended,
+  },
+  {
+    name: 'Cypress overrides',
+    files: TEST_FILES,
     rules: {
       'cypress/no-unnecessary-waiting': 'off',
       'cypress/unsafe-to-chain-command': 'off',
-      'mocha/no-setup-in-describe': 'off',
+    },
+  },
+  {
+    name: 'Generic test overrides',
+    files: TEST_FILES,
+    rules: {
       'no-console': 'warn',
       'no-debugger': 'warn',
```

Comment:
Do we still need this section if these rules are included in the rules for `Base globals & quality rules`?